### PR TITLE
Support Shibboleth in KohaRest driver

### PR DESCRIPTION
**Pull request by PerplexedTheta**
Hello again,

Following on from #3259 — I have recreated the branch. Had no idea pull bot was a thing; what an infuriating feature! Alas, I think I've clobbered it, now.

Many thanks!

### CODE REVIEW

**Comment by demiankatz** [comment]
Thanks, @Pilvedore! Adding strangeness to strangeness, I'm not entirely sure why #3259 shows as merged, though I guess when Pullbot force-pushed commits from the dev branch, GitHub noticed that those commits were already on the dev branch and auto-closed it. Anyway, I'm glad you were able to get this straightened out. I've requested a review from @EreMaijala so we can pick up where we left off on the previous PR.

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: Should this be `??` rather than `||`? I think we want to default to `''` if `$password` is null, but we might not want to for other falsy values -- i.e. what if your password is `00000000`?

**Review by demiankatz — CHANGES_REQUESTED** [review]
One small question...

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php
Comment (PerplexedTheta) [reply module/VuFind/src/VuFind/ILS/Driver/KohaRest.php]: I think you've got a good point, well caught - I'll have a quick test and push up a commit when I'm happy

**Comment by PerplexedTheta** [comment]
> Should this be ?? rather than ||? I think we want to default to '' if $password is null, but we might not want to for other falsy values -- i.e. what if your password is 00000000?

This is now honoured in the code

**Review by demiankatz — APPROVED** [review]
Thanks for the fix, @Pilvedore. This looks good to me, so I'm approving it... but since I'm not expert on the Koha aspects, I'll wait for @EreMaijala's final review before merging.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Actually, I take it back! phpstan is complaining that $password can never be null, so the new code is pointless. Maybe the safest implementation is just `(string)$password` to be sure we always get a string value.

**Comment by PerplexedTheta** [comment]
@demiankatz yes, good solution once again. I didn't realise PHP was typed as tightly as that. Added that into the code now — hopefully we see a CI pass shortly.

Thanks again!

**Comment by demiankatz** [comment]
@Pilvedore, thanks for the latest adjustments. However, it looks like the `|| ''` got reintroduced along the way, which takes us back to my initial concerns about that code. I've just gone ahead and deleted that clause, since the `(string)` cast should guarantee that a non-string empty value gets typecast to `''` anyway. Please let me know if you have any concerns about this, but I think it should be okay now!

**Comment by demiankatz** [comment]
Merging with @EreMaijala's blessing. Thanks, @Pilvedore!

**Review by demiankatz — APPROVED** [review]

**Pull request merged**
